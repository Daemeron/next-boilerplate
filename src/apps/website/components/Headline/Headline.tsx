import React, { ReactChild } from 'react';
import styled from 'styled-components';

import { ThemePropsInterface } from '../../core/theme/Theme';

interface HeadlineProps {
  tag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  children?: ReactChild | ReactChild[];
}

export const Headline = (props: HeadlineProps): JSX.Element => (
  <GenricHeadline as={props.tag ? props.tag : 'h6'}>{props.children}</GenricHeadline>
);

export const GenricHeadline = styled.h1<ThemePropsInterface>`
  font-size: 2rem;
  line-height: 2.4rem;
`;
