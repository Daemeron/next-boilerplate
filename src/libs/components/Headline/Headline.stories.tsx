import React, { Fragment } from 'react';

import { Headline } from './Headline';

export default {
  title: 'Components/Headline',
};

export const WithText = (): JSX.Element => (
  <Fragment>
    <Headline>dummy text</Headline>
  </Fragment>
);
