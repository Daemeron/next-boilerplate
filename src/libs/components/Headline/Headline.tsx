import React, { ReactChild } from 'react';
import styled from 'styled-components';

interface HeadlineProps {
  tag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  children?: ReactChild | ReactChild[];
}

export const Headline = (props: HeadlineProps): JSX.Element => (
  <GenricHeadline as={props.tag ? props.tag : 'h6'}>{props.children}</GenricHeadline>
);

export const GenricHeadline = styled.h1``;
