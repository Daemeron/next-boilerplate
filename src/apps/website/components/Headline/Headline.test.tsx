import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Headline } from './Headline';

describe('Headline', () => {
  test('matches snapshot', () => {
    const headline = render(<Headline>test</Headline>);
    expect(headline.container).toMatchSnapshot();
  });

  test('sets initial font-size and line-height', () => {
    const headline = render(<Headline>test</Headline>);
    expect(headline.container.firstChild).toHaveStyleRule('font-size', '2rem');
    expect(headline.container.firstChild).toHaveStyleRule('line-height', '2.4rem');
  });

  test('renders single child', () => {
    const headline = render(<Headline>test</Headline>);
    expect(headline.queryByText('test')).toBeTruthy();
  });

  test('renders multiple children', () => {
    const headline = render(
      <Headline>
        <span>test</span>
        <span>test</span>
      </Headline>
    );
    expect(headline.queryAllByText('test').length).toEqual(2);
  });
});
