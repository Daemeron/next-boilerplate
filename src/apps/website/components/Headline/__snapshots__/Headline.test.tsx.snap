// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline matches snapshot 1`] = `
.c0 {
  font-size: 2rem;
  line-height: 2.4rem;
}

<div>
  <h6
    class="c0"
  >
    test
  </h6>
</div>
`;
